src/aleph/tcp.clj

```diff
@@ -24,14 +24,16 @@
 (alter-meta! #'->TcpConnection assoc :private true)
 
 (defn- ^ChannelHandler server-channel-handler
-  [handler {:keys [raw-stream?] :as options}]
+  [handler {:keys [raw-stream? exception-handler] :as options}]
   (let [in (atom nil)]
     (netty/channel-inbound-handler
 
       :exception-caught
       ([_ ctx ex]
-        (when-not (instance? IOException ex)
-          (log/warn ex "error in TCP server")))
+       (if (some? exception-handler)
+         (exception-handler @in ex)
+         (when-not (instance? IOException ex)
+           (log/warn ex "error in TCP server"))))
```

Comment:
Why only log if its an IOException? 
Also, wouldnt it be better if there was a default handler? 
Something like, 
```clojure
(or exception-handler #(log/warn ex "error in TCP server"))))
```

Reply:
@vigneshsarma it was the previous behavior which I would rather leave untouched. It actually logs when the exception is not an IO one